src/gui/sdlmain.cpp

```diff
@@ -2309,9 +2383,17 @@ static void setup_window_sizes_from_conf(const char *windowresolution_val,
 	// Save the coarse bounds in the SDL struct for future sizing events
 	sdl.desktop.requested_window_bounds = {coarse_size.x, coarse_size.y};
 
-	// Refine the coarse resolution and save it in the SDL struct
-	const auto refined_size = refine_window_size(coarse_size, refined_scaling_mode,
-	                                             wants_stretched_pixels);
+	// Refine the coarse resolution and save it in the SDL struct.
+	auto refined_size = coarse_size;
+	if (window_resolution_specified && sdl.restrict_resolution) {
+		// If viewport resolution restriction is enabled, the refinement is
+		// applied to the restricted resolution instead of the the window dimensions.
+		refined_size = clamp_to_minimum_window_dimensions(coarse_size);
+	} else {
+		refined_size = refine_window_size(coarse_size,
+										  refined_scaling_mode,
+										  wants_stretched_pixels);
+	}
```

Comment:
Some tabs got inserted here for the indented 2nd and 3rd arguments (instead of spacing into the aligned column).  Using the `./script/format-commit.sh` script should fix it up.

Reply:
Done.